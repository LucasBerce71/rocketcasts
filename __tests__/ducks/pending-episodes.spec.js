import ActionCreators, { reducer, INITIAL_STATE } from 'store/ducks/pending-episodes';

test('request', () => {
  const state = reducer(INITIAL_STATE, ActionCreators.pendingRequest());

  expect(state).toMatchSnapshot();
});

test('success', () => {
  const state = reducer(INITIAL_STATE, ActionCreators.pendingSuccess(['result']));

  expect(state).toMatchSnapshot();
});

test('failure', () => {
  const state = reducer(INITIAL_STATE, ActionCreators.pendingFailure());

  expect(state).toMatchSnapshot();
});

test('refresh', () => {
  const state = reducer(INITIAL_STATE, ActionCreators.pendingRefresh());

  expect(state).toMatchSnapshot();
});
