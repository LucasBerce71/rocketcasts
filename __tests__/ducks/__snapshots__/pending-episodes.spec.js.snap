// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`failure 1`] = `
Object {
  "data": Array [],
  "error": true,
  "loading": false,
  "refreshing": false,
}
`;

exports[`refresh 1`] = `
Object {
  "data": Array [],
  "error": false,
  "loading": false,
  "refreshing": true,
}
`;

exports[`request 1`] = `
Object {
  "data": Array [],
  "error": null,
  "loading": true,
  "refreshing": false,
}
`;

exports[`success 1`] = `
Object {
  "data": Array [
    "result",
  ],
  "error": null,
  "loading": false,
  "refreshing": false,
}
`;
